### PR TITLE
feat: handling the vertex_removed event

### PR DESCRIPTION
**Pull request by andreabadesso**
### Motivation

- We should handle the `VERTEX_REMOVED` event
- We should have balance integration tests for the `invalid_mempool` scenario
- We should fix multiple db methods that were failing on empty params

### Acceptance Criteria

- Include here all things that this PR should solve

### Checklist
- [ ] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged
- [ ] Make sure either the unit tests and/or the QA tests are capable of testing the new features
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: packages/daemon/src/db/index.ts [hunk 1]
```diff
@@ -497,6 +497,10 @@ export const updateAddressTablesWithTx = async (
   timestamp: number,
   addressBalanceMap: StringMap<TokenBalanceMap>,
 ): Promise<void> => {
+  if (Object.keys(addressBalanceMap).length === 0) {
+    // No need to do anything here, the transaction most likely has no inputs and outputs
```
Comment (glevco) [inline packages/daemon/src/db/index.ts]: "most likely"? Is it possible that this is not the case?

**Review by glevco — DISMISSED** [review]

File: packages/daemon/src/db/index.ts (hunk 1 above)
Comment (andreabadesso) [reply packages/daemon/src/db/index.ts]: No.

Removed the comment

Comment (glevco) [reply packages/daemon/src/db/index.ts]: Did you push it?

**Review by r4mmer — DISMISSED** [review]

File: packages/daemon/src/db/index.ts (hunk 1 above)
Comment (andreabadesso) [reply packages/daemon/src/db/index.ts]: Ops, I pushed it to the invalid scripts branch, just cherry-picked it, can you check again?

**Review by r4mmer — APPROVED** [review]

**Review by glevco — APPROVED** [review]

**Pull request merged**